pkg/bloomcompactor/mergecompactor.go

```diff
@@ -108,42 +113,57 @@ func createPopulateFunc(ctx context.Context, job Job, storeClient storeClient, b
 	}
 }
 
-func mergeCompactChunks(logger log.Logger,
+func mergeCompactChunks(
+	logger log.Logger,
 	populate func(*v1.Series, *v1.Bloom) error,
 	mergeBlockBuilder *PersistentBlockBuilder,
-	blockIters []v1.PeekingIterator[*v1.SeriesWithBloom], seriesIter *v1.SliceIter[*v1.Series],
-	job Job) (bloomshipper.Block, error) {
+	blockIters []v1.PeekingIterator[*v1.SeriesWithBloom],
+	seriesIter v1.PeekingIterator[*v1.Series],
+	job Job,
+) ([]bloomshipper.Block, error) {
 
 	mergeBuilder := v1.NewMergeBuilder(
 		blockIters,
 		seriesIter,
 		populate)
 
-	checksum, err := mergeBlockBuilder.mergeBuild(mergeBuilder)
-	if err != nil {
-		level.Error(logger).Log("msg", "failed merging the blooms", "err", err)
-		return bloomshipper.Block{}, err
-	}
-	data, err := mergeBlockBuilder.Data()
-	if err != nil {
-		level.Error(logger).Log("msg", "failed reading bloom data", "err", err)
-		return bloomshipper.Block{}, err
-	}
+	// TODO(salvacorts): Reuse buffer
+	mergedBlocks := make([]bloomshipper.Block, 0)
 
-	mergedBlock := bloomshipper.Block{
-		BlockRef: bloomshipper.BlockRef{
-			Ref: bloomshipper.Ref{
-				TenantID:       job.tenantID,
-				TableName:      job.tableName,
-				MinFingerprint: uint64(job.minFp),
-				MaxFingerprint: uint64(job.maxFp),
-				StartTimestamp: job.from,
-				EndTimestamp:   job.through,
-				Checksum:       checksum,
+	for {
+		// Merge/Create blocks until there are no more series to process
+		if _, hasNext := seriesIter.Peek(); !hasNext {
+			break
+		}
+
+		blockPath, checksum, bounds, err := mergeBlockBuilder.mergeBuild(mergeBuilder)
+		if err != nil {
+			level.Error(logger).Log("msg", "failed merging the blooms", "err", err)
+			return []bloomshipper.Block{}, err
+		}
+		data, err := mergeBlockBuilder.Data()
+		if err != nil {
+			level.Error(logger).Log("msg", "failed reading bloom data", "err", err)
+			return []bloomshipper.Block{}, err
```

Comment:
could you please also wrap the errors? it's super useful to have some kinda stack trace for the error:
example:
```suggestion
			return []bloomshipper.Block{}, fmt.Errorf("failed reading bloom data: %w", err)
```

Reply:
Done :)